ads/baidu.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  loadScript,
+  validateData,
+} from '../3p/3p';
+
+/**
+ * @param {!Window} global
+ * @param {!Object} data
+ */
+export function baidu(global, data) {
+  validateData(data, ['cproid']);
+
+  const id = '_' + Math.random().toString(36).slice(2);
+  const container = global.document.createElement('div');
+  container.id = id;
+  global.document.getElementById('c').appendChild(container);
+
+  global.slotbydup = global.slotbydup || [];
+  global.slotbydup.push({
+    id: data['cproid'],
+    container: id,
+    display: 'inlay-fix',
+    async: true,
+  });
+
+  loadScript(
+      global,
+      'https://dup.baidustatic.com/js/dm.js',
+      () => {
+        global.context.renderStart();
+      },
+      () => {
+        global.context.noContentAvailable();
```

Comment:
this only gets called when the script is failed to load.

does dm.js script fire global events when ad response is back? that will be the time to call renderStart or noContentAvailable, depending on if the ad is filled or not.